pkg/controller/applicationlayer/applicationlayer_controller.go

```diff
@@ -304,16 +304,25 @@ func (r *ReconcileApplicationLayer) Reconcile(ctx context.Context, request recon
 
 // updateApplicationLayerWithDefaults populates the applicationlayer with defaults.
 func updateApplicationLayerWithDefaults(al *operatorv1.ApplicationLayer) {
-	defaultLogIntervalSeconds := int64(5)
-	defaultLogRequestsPerInterval := int64(-1)
+	var defaultLogIntervalSeconds int64 = 5
+	var defaultLogRequestsPerInterval int64 = -1
+	var defaultLogCollectionDisabled operatorv1.LogCollectionStatusType = "Disabled"
+	var defaultWebApplicationFirewallDisabled operatorv1.WAFStatusType = "Disabled"
 
-	if al.Spec.LogCollection != nil {
-		if al.Spec.LogCollection.LogRequestsPerInterval == nil {
-			al.Spec.LogCollection.LogRequestsPerInterval = &defaultLogRequestsPerInterval
-		}
-		if al.Spec.LogCollection.LogIntervalSeconds == nil {
-			al.Spec.LogCollection.LogIntervalSeconds = &defaultLogIntervalSeconds
-		}
+	if al.Spec.LogCollection == nil {
+		al.Spec.LogCollection = new(operatorv1.LogCollectionSpec)
+	}
+	if al.Spec.LogCollection.CollectLogs == nil {
+		al.Spec.LogCollection.CollectLogs = &defaultLogCollectionDisabled
+	}
+	if al.Spec.LogCollection.LogRequestsPerInterval == nil {
```

Comment:
It's a bit odd to default a feature to "off" and then default other fields related to it to be "on". I would say this is probably an invalid configuration - you're basically saying "Disable log collection, and also produce this many logs per interval".

For context, if we didn't have an `enabled` field already, I would think the correct idiomatic way to do this would be:

```yaml
# Enable log collection
spec:
  logCollection:
    logIntervalSeconds: 5
```

```yaml
# disable log collection
spec: 
  logCollection:
    logIntervalSeconds: 0
``` 

But we do already have an `enabled` field, and so we just need to make sure that the "enabled" field is consistent with the others.

So, this would be invalid:

```yaml
spec:
  logCollection:
    enabled: true
    logIntervalSeconds: 0
```

And so would this:

```yaml
spec:
  logCollection:
    enabled: false
    logInterval Seconds: 5
```

because it's two pieces of conflicting information given to us by the user.

Reply:
I wouldn't agree with this statement:
`"Disable log collection, and also produce this many logs per interval"`
It is a legitimate use case to have some configuration working and to disable it without loosing the previous values that were set.

This code is about L7 log collection - I couldn't find a doc that would be descriptive of what an expected behavior in this case should be. I'm sure the default values - that aren't new in this context here - \ have a meaning for that feature also.

Reply:
> It is a legitimate use case to have some configuration working and to disable it without loosing the previous values that were set.

Not via a Kubernetes API it isn't. 

The way to meet this use-case in a Kubernetes world is like so:

```
kubectl get applicationlayer default > application-layer.backup
```

Then to revert any changes:
```
kubectl replace -f application-layer.backup
```

Reply:
^ my point is that this is the model we have followed for all of our other APIs. There's no "create an APIServer but don't actually run the apiserver" option, etc., etc. We should be consistent across all of our APIs, and unless I'm mistaken there's nothing special about WAF that makes this an exception. 

It's an anti-pattern to have the "enabled" fields at all in my opinion, but following the above at least makes it consistent with how the structure would have worked if we hadn't introduced the "enabled" fields.

Reply:
Done, as per our conversation.